src/test/java/seedu/letsgethired/logic/commands/NoteDeleteCommandTest.java

```diff
@@ -0,0 +1,75 @@
+package seedu.letsgethired.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.letsgethired.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.letsgethired.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.letsgethired.testutil.TypicalIndexes.INDEX_FIRST_APPLICATION;
+import static seedu.letsgethired.testutil.TypicalIndexes.INDEX_SECOND_APPLICATION;
+import static seedu.letsgethired.testutil.TypicalInternApplications.getTypicalInternApplications;
+import static seedu.letsgethired.testutil.TypicalInternApplications.getTypicalInternTracker;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.letsgethired.commons.core.index.Index;
+import seedu.letsgethired.logic.Messages;
+import seedu.letsgethired.model.InternTracker;
+import seedu.letsgethired.model.Model;
+import seedu.letsgethired.model.ModelManager;
+import seedu.letsgethired.model.UserPrefs;
+import seedu.letsgethired.model.application.InternApplication;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for NoteInsertCommand.
+ */
+public class NoteDeleteCommandTest {
+
+    private Model model = new ModelManager(getTypicalInternTracker(), new UserPrefs());
+
+    @Test
+    public void execute_invalidInternApplicationIndexUnfilteredList_failure() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredInternApplicationList().size() + 1);
+        NoteDeleteCommand noteCommand = new NoteDeleteCommand(outOfBoundIndex, 0);
+
+        assertCommandFailure(noteCommand, model, Messages.MESSAGE_INVALID_INTERN_APPLICATION_DISPLAYED_INDEX);
+    }
+
```

Comment:
Perhaps we should have an additional test for command failure from invalid note index?

Reply:
Done